app/src/main/java/com/elavon/converge/processor/ConvergeService.java

```diff
@@ -0,0 +1,123 @@
+package com.elavon.converge.processor;
+
+import android.util.Log;
+
+import com.elavon.converge.exception.ConvergeClientException;
+import com.elavon.converge.model.ElavonTransactionRequest;
+import com.elavon.converge.model.ElavonTransactionResponse;
+import com.elavon.converge.model.ElavonTransactionSearchRequest;
+import com.elavon.converge.model.ElavonTransactionSearchResponse;
+import com.elavon.converge.model.mapper.ConvergeMapper;
+
+import java.math.BigDecimal;
+import java.net.SocketTimeoutException;
+import java.util.Date;
+
+import javax.inject.Inject;
+
+public class ConvergeService {
+
+    protected final ConvergeClient convergeClient;
+    protected final int maxRetryCount;
+
+    @Inject
+    public ConvergeService(final ConvergeClient convergeClient, final int maxRetryCount) {
+        this.convergeClient = convergeClient;
+        this.maxRetryCount = maxRetryCount;
+    }
+
+    public void create(final ElavonTransactionRequest request, final ConvergeCallback<ElavonTransactionResponse> callback) {
+        create(request, callback, 0, new Date());
+    }
+
+    private void create(
+            final ElavonTransactionRequest request,
+            final ConvergeCallback<ElavonTransactionResponse> callback,
+            final int retryCount,
+            final Date originalTime) {
+
+        if (retryCount > maxRetryCount) {
+            // TODO submit reversal
+            Log.e("ConvergeService", "Max retry count reached. Starting reversal...");
+            callback.onFailure(new ConvergeClientException("Transaction failed after all retries"));
+            return;
+        }
+        Log.i("ConvergeService", "Create with retry count: " + retryCount);
+
+        final ConvergeCallback cb = new ConvergeCallback<ElavonTransactionResponse>() {
+            @Override
+            public void onResponse(final ElavonTransactionResponse response) {
+                callback.onResponse(response);
+            }
+
+            @Override
+            public void onFailure(final Throwable t) {
+                // retry if timeout
+                if (t instanceof SocketTimeoutException) {
+                    Log.e("ConvergeService", "Create transaction timeout. Retrying...");
+                    checkAndCreate(request, callback, retryCount + 1, originalTime);
```

Comment:
retry only when timeout

Reply:
oh forgot to ask - do they have a way to enforce idempotency ?

Reply:
to prevent duplicate charges